crates/router/src/core/admin.rs

```diff
@@ -831,74 +816,234 @@ pub async fn create_business_profile_from_business_labels(
-    state: SessionState,
-    merchant_account_update: api::MerchantAccountUpdate,
-    merchant_id: id_type::MerchantId,
-) -> RouterResult<()> {
-    if merchant_account_update.return_url.is_some()
-        || merchant_account_update.webhook_details.is_some()
-        || merchant_account_update
-            .enable_payment_response_hash
-            .is_some()
-        || merchant_account_update
-            .redirect_to_merchant_with_http_post
-            .is_some()
-    {
-        // Update these fields in all the business profiles
-        let business_profiles = state
-            .store
-            .list_business_profile_by_merchant_id(&merchant_id)
-            .await
-            .to_not_found_response(errors::ApiErrorResponse::BusinessProfileNotFound {
-                id: merchant_id.get_string_repr().to_owned(),
-            })?;
+#[cfg(any(feature = "v1", feature = "v2", feature = "olap"))]
+#[async_trait::async_trait]
+trait MerchantConnectorAccountUpdateBridge {
+    async fn get_update_merchant_object(
+        self,
+        state: &SessionState,
+        merchant_id: &id_type::MerchantId,
+        key_store: &domain::MerchantKeyStore,
+    ) -> RouterResult<storage::MerchantAccountUpdate>;
+}
 
-        let business_profile_update = admin_types::BusinessProfileUpdate {
-            profile_name: None,
-            return_url: merchant_account_update.return_url,
-            enable_payment_response_hash: merchant_account_update.enable_payment_response_hash,
-            payment_response_hash_key: merchant_account_update.payment_response_hash_key,
-            redirect_to_merchant_with_http_post: merchant_account_update
-                .redirect_to_merchant_with_http_post,
-            webhook_details: merchant_account_update.webhook_details,
-            metadata: None,
-            routing_algorithm: None,
+#[cfg(all(
+    any(feature = "v1", feature = "v2"),
+    not(feature = "merchant_account_v2")
+))]
+#[async_trait::async_trait]
+impl MerchantConnectorAccountUpdateBridge for api::MerchantAccountUpdate {
+    async fn get_update_merchant_object(
+        self,
+        state: &SessionState,
+        merchant_id: &id_type::MerchantId,
+        key_store: &domain::MerchantKeyStore,
+    ) -> RouterResult<storage::MerchantAccountUpdate> {
+        use common_utils::ext_traits::ConfigExt;
+
+        let key_manager_state = &state.into();
+        let key = key_store.key.get_inner().peek();
+
+        let db = state.store.as_ref();
+
+        if &self.merchant_id != merchant_id {
+            Err(report!(errors::ValidationError::IncorrectValueProvided {
+                field_name: "parent_merchant_id"
+            })
+            .attach_printable(
+                "If `sub_merchants_enabled` is true, then `parent_merchant_id` is mandatory",
+            )
+            .change_context(errors::ApiErrorResponse::InvalidDataValue {
+                field_name: "parent_merchant_id",
+            }))?;
+        }
```

Comment:
Is this check relevant: we're checking against merchant ID, but raising an error about `parent_merchant_id`?